### PR TITLE
8305787: Wrong debugging information printed with TraceOptoOutput

### PR DESCRIPTION
**Pull request by jaskarth**
This patch fixes a minor bug in aldc where the wrong resource names are printed when the flag TraceOptoOutput is enabled to debug instruction scheduling.
As an example, the output:
```
*** Bundle: 1 instr, resources: D0 BR
 126 salI_rReg_imm === _ 240 |271 [[ 127 125 ]] #5/0x00000005
```
states that the bundle is using resources D0 and BR, but the second resource used is actually ALU0.

The issue is caused because `pipeline->_rescount` is only incremented for discrete resources [(here)](https://github.com/openjdk/jdk/blob/master/src/hotspot/share/adlc/adlparse.cpp#L1612), resources specified without `=`. However, the list of names is added to for *all* resources [(here)](https://github.com/openjdk/jdk/blob/master/src/hotspot/share/adlc/adlparse.cpp#L1652), so using `_rescount` to index the names causes it to go out of sync. The fix is found in [output_h.cpp](https://github.com/openjdk/jdk/blob/master/src/hotspot/share/adlc/output_h.cpp#L2231), where it uses the iterator to go through all the resources and use only the ones that are discrete. I applied that fix to this case, and also fixed the other instances of this bug. Reviews on this fix would be appreciated!

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (2 reviews required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer), 1 [Author](https://openjdk.org/bylaws#author))

### Issue
 * [JDK-8305787](https://bugs.openjdk.org/browse/JDK-8305787): Wrong debugging information printed with TraceOptoOutput

### Reviewers
 * [Tobias Hartmann](https://openjdk.org/census#thartmann) (@TobiHartmann - **Reviewer**) ⚠️ Review applies to [79e3f744](https://git.openjdk.org/jdk/pull/13403/files/79e3f74453c8030adb06445ac871da06d893eab3)
 * [Vladimir Kozlov](https://openjdk.org/census#kvn) (@vnkozlov - **Reviewer**) ⚠️ Review applies to [79e3f744](https://git.openjdk.org/jdk/pull/13403/files/79e3f74453c8030adb06445ac871da06d893eab3)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/13403/head:pull/13403` \
`$ git checkout pull/13403`

Update a local copy of the PR: \
`$ git checkout pull/13403` \
`$ git pull https://git.openjdk.org/jdk.git pull/13403/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 13403`

View PR using the GUI difftool: \
`$ git pr show -t 13403`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/13403.diff">https://git.openjdk.org/jdk/pull/13403.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/13403#issuecomment-1501357853)

### CODE REVIEW

**Review by TobiHartmann — APPROVED** [review]
Looks reasonable to me but I'm not an expert in this code. Another review would be good.

File: src/hotspot/share/adlc/formsopt.cpp [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 1998, 2022, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 1998, 2023, Oracle and/or its affiliates. All rights reserved.
```
Comment (vnkozlov) [inline src/hotspot/share/adlc/formsopt.cpp:2]: You mistakenly updated .cpp file instead of formsopt.hpp

**Review by vnkozlov — APPROVED** [review]
I agree with this fix. Only one comment about copyright line update.

**Comment by jaskarth** [comment]
Thanks a lot for the reviews @TobiHartmann and @vnkozlov! I've updated the PR with the copyright line fixed.

**Comment by vnkozlov** [comment]
I submitted our internal testing before sponsoring.

**Comment by vnkozlov** [comment]
My testing passed.

**Pull request closed without merging**